---
output: html_document
---

## Projecting scRNA-seq data

```{r echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE}
library(knitr)
library(googleVis)
opts_chunk$set(fig.align="center", dev = 'png')
op <- options(gvis.plot.tag='chart')
```
```{r, echo=TRUE, message=FALSE, warning=FALSE}
library(scmap)
library(scater)
library(SingleCellExperiment)
set.seed(1234567)
```

As more and more scRNA-seq datasets become available, carrying out comparisons between them is key. A central application is to compare datasets of similar biological origin collected by different labs to ensure that the annotation and the analysis is consistent. Moreover, as very large references, e.g. the Human Cell Atlas (HCA) [@Regev2017-mw], become available, an important application will be to project cells from a new sample (e.g. from a disease tissue) onto the reference to characterize differences in composition, or to detect new cell-types. Conceptually, such projections are similar to the popular BLAST method [@Altschul1990-ts], which makes it possible to quickly find the closest match in a database for a newly identified nucleotide or amino acid sequence. 

We recently developed `scmap` [@Kiselev2017-nb] - a method for projecting cells from a scRNA-seq experiment onto the cell-types identified in other experiments. Additionally, a cloud version of `scmap` can be run for free, without restrictions, from [http://www.hemberg-lab.cloud/scmap](http://www.hemberg-lab.cloud/scmap).

### Datasets

We will run `scmap` on two human pancreas datasets: [@Muraro2016-yk] and [@Segerstolpe2016-wc]. Since the pancreas has been widely studied, these datasets are well annotated. As usual cell type labels are stored in the `cell_type1` column of the `colData` slots.
```{r}
muraro <- readRDS("pancreas/muraro.rds")
segerstolpe <- readRDS("pancreas/segerstolpe.rds")
```

`scmap` can be used to perform both types of projections to either a single dataset or to a reference collection of datasets (Atlas):

```{r, echo=FALSE, out.width = '80%', fig.cap="scmap can be used to compare the annotations of two different samples by providing a one to one mapping between the cells."}
knitr::include_graphics("figures/scmap1.png")
```

```{r, echo=FALSE, out.width = '80%', fig.cap=" scmap can also be used to project cells from a new experiment onto an annotated reference."}
knitr::include_graphics("figures/scmap2.png")
```

### Feature Selection
Once we have a `SingleCellExperiment` object we can run `scmap`. Firstly, we need to select the most informative features from our input dataset:
```{r}
muraro <- selectFeatures(muraro, suppress_plot = FALSE)
```

Features highlighted with the red colour will be used in the futher analysis (projection).

Features are stored in the `scmap_features` column of the `rowData` slot of the input object. By default scmap selects 500 features (it can also be controlled by setting `n_features` parameter):
```{r}
table(rowData(muraro)$scmap_features)
```

### `scmap-cluster`

The `scmap-cluster` index of a reference dataset is created by finding the median gene expression for each cluster. By default scmap uses the `cell_type1` column of the `colData` slot in the reference to identify clusters. Other columns can be manually selected by adjusting `cluster_col` parameter:
```{r}
muraro <- indexCluster(muraro)
```

The function indexCluster automatically writes the scmap_cluster_index item of the metadata slot of the reference dataset.
```{r}
head(metadata(muraro)$scmap_cluster_index)
```

One can also visualise the index:
```{r}
heatmap(as.matrix(metadata(muraro)$scmap_cluster_index))
```

#### Projecting

Once the `scmap-cluster` index has been generated we can use it to project `segerstolpe` dataset to `muraro` dataset. This can be done with one index at a time, but `scmap` also allows for simultaneous projection to multiple indexes if they are provided as a list:

```{r}
scmapCluster_results <- scmapCluster(
  projection = segerstolpe, 
  index_list = list(
    muraro = metadata(muraro)$scmap_cluster_index
  )
)
```

### Results

`scmap-cluster` projects the query dataset to all projections defined in the index_list. The results of cell label assignements are merged into one matrix:
```{r}
head(scmapCluster_results$scmap_cluster_labs)
```

Corresponding similarities are stored in the scmap_cluster_siml item:
```{r}
head(scmapCluster_results$scmap_cluster_siml)
```

`scmap` also provides combined results of all reference dataset (choose labels corresponding to the largest similarity across reference datasets):
```{r}
head(scmapCluster_results$combined_labs)
```

Clearly the projection is almost perfect. With `scmap` one can also plot a [Sankey diagram](https://developers.google.com/chart/interactive/docs/gallery/sankey) (however, `cell_type1` columns have to be provided in the `colData` slots of both the reference and the projection datasets):
```{r results='asis', tidy=FALSE}
plot(
  getSankey(
    colData(segerstolpe)$cell_type1, 
    scmapCluster_results$scmap_cluster_labs[,'muraro'],
    plot_height = 400
  )
)
```

### sessionInfo()

```{r echo=FALSE}
sessionInfo()
```


